internal/apis/storage/volume_types.go

```diff
@@ -122,8 +122,24 @@ type VolumeList struct {
 	Items []Volume
 }
 
+// ReclaimPolicy for a Volume tells what to do with the volume after its managing resource has been deleted
+type ReclaimPolicy string
+
+const (
+	// The Volume resource is not deleted after the managing resource has been deleted
+	Retain ReclaimPolicy = "Retain"
+	// The Volume resource is garbage-collected when the managing resource has been deleted
+	Delete ReclaimPolicy = "Delete"
+)
+
+type EphemeralVolumeSpec struct {
+	VolumeSpec
+	// ReclaimPolicy defines the strategy for the corresponding volume resource after its managing resource has been deleted
+	ReclaimPolicy ReclaimPolicy
+}
+
 // VolumeTemplateSpec is the specification of a Volume template.
 type VolumeTemplateSpec struct {
 	metav1.ObjectMeta
-	Spec VolumeSpec
+	Spec EphemeralVolumeSpec
```

Comment:
Minor: please move this upwards in between `type VolumeAccess struct` and `type VolumeStatus struct` at line https://github.com/ironcore-dev/ironcore/blob/987deb12cd36ba28f1cbaa9da821df2885e70db3/api/storage/v1alpha1/volume_types.go#L58